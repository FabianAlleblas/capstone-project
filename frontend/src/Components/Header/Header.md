Header: Primary

```js
<Header>My Plants</Header>
```

Header: Secondary

```js
<Header secondaryStyle>Add Plant</Header>
```
