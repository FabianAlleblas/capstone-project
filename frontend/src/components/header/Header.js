import styled from 'styled-components/macro'

export default function Header() {
  return (
    <HeaderStyled>
      <HeadingContainer>
        <Heading>My Plants</Heading>
      </HeadingContainer>
    </HeaderStyled>
  )
}

const HeaderStyled = styled.header`
  display: flex;
`

const HeadingContainer = styled.div`
  background: var(--primary-dark);
  padding: 20px 30px 13px 20px;
  border-radius: 0 0 50px 0;
  box-shadow: 0 0 4px var(--shadow);
`

const Heading = styled.h1`
  color: var(--primary-light);
  font-size: 1.75rem;
  font-weight: 700;
  line-height: 1;
`
