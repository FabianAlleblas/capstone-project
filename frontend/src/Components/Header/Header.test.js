import { render } from '@testing-library/react'
import 'jest-styled-components'
import Header from './Header'

describe('Header', () => {
  it('reders with the correct text and logout button', () => {
    const { getByText, getByRole, container } = render(
      <Header showLogoutButton>My Plants</Header>
    )

    expect(getByText('My Plants')).toBeInTheDocument()
    expect(getByRole('button', { name: /LogoutIcon/i })).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
  it('reders with the back and edit button', () => {
    const { getByRole, container } = render(
      <Header showBackButton showEditButton>
        Helga
      </Header>
    )

    expect(getByRole('button', { name: /ArrowIcon/i })).toBeInTheDocument()
    expect(getByRole('button', { name: /EditIcon/i })).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('reders with the delete button', () => {
    const { getByRole, container } = render(
      <Header showDeleteButton>Edit your Plant</Header>
    )

    expect(getByRole('button', { name: /DeleteIcon/i })).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
  it('renders the header without a box-shadow', () => {
    const { getByTestId, container } = render(
      <Header secondaryStyle>Add plant</Header>
    )

    expect(getByTestId('heading-container')).not.toHaveStyle(
      'box-shadow: 0 0 4px var(--shadow)'
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
