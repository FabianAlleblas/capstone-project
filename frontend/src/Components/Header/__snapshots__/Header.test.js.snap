// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header reders with the back and edit button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  z-index: 10;
}

.c1 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  background: var(--primary-dark);
  border-radius: 0 0 50px 0;
  box-shadow: 0 0 4px var(--shadow);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 20px 30px 13px 20px;
}

.c4 {
  color: var(--primary-light);
  font-size: 1.75rem;
  font-weight: 700;
  line-height: 1;
}

.c3 {
  margin-right: 12px;
}

.c5 {
  margin-right: 12px;
}

.c2 {
  background-color: transparent;
  border: none;
}

<header
  class="c0"
>
  <div
    class="c1"
    data-testid="heading-container"
  >
    <button
      class="c2"
    >
      <svg
        class="c3"
      >
        arrowicon.svg
      </svg>
    </button>
    <h1
      class="c4"
    >
      Helga
    </h1>
  </div>
  <button
    class="c2"
  >
    <svg
      class="c5"
    >
      editicon.svg
    </svg>
  </button>
</header>
`;

exports[`Header reders with the correct text and logout button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  z-index: 10;
}

.c1 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  background: var(--primary-dark);
  border-radius: 0 0 50px 0;
  box-shadow: 0 0 4px var(--shadow);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 20px 30px 13px 20px;
}

.c2 {
  color: var(--primary-light);
  font-size: 1.75rem;
  font-weight: 700;
  line-height: 1;
}

.c4 {
  margin-right: 8px;
}

.c3 {
  background-color: transparent;
  border: none;
}

<header
  class="c0"
>
  <div
    class="c1"
    data-testid="heading-container"
  >
    <h1
      class="c2"
    >
      My Plants
    </h1>
  </div>
  <button
    class="c3"
  >
    <svg
      class="c4"
    >
      logouticon.svg
    </svg>
  </button>
</header>
`;

exports[`Header reders with the delete button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  z-index: 10;
}

.c1 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  background: var(--primary-dark);
  border-radius: 0 0 50px 0;
  box-shadow: 0 0 4px var(--shadow);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 20px 30px 13px 20px;
}

.c2 {
  color: var(--primary-light);
  font-size: 1.75rem;
  font-weight: 700;
  line-height: 1;
}

.c4 {
  margin-right: 12px;
}

.c3 {
  background-color: transparent;
  border: none;
}

<header
  class="c0"
>
  <div
    class="c1"
    data-testid="heading-container"
  >
    <h1
      class="c2"
    >
      Edit your Plant
    </h1>
  </div>
  <button
    class="c3"
  >
    <svg
      class="c4"
    >
      deleteicon.svg
    </svg>
  </button>
</header>
`;

exports[`Header renders the header without a box-shadow 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  z-index: 10;
}

.c1 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  background: var(--primary-dark);
  border-radius: 0 0 50px 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 20px 30px 13px 20px;
}

.c2 {
  color: var(--primary-light);
  font-size: 1.75rem;
  font-weight: 700;
  line-height: 1;
}

<header
  class="c0"
>
  <div
    class="c1"
    data-testid="heading-container"
  >
    <h1
      class="c2"
    >
      Add plant
    </h1>
  </div>
</header>
`;
